### PR TITLE
[WIP][CatalogPromotion] fix order of CP labels

### PR DESCRIPTION
**Pull request by AdamKasp**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| License         | MIT

### CODE REVIEW

File: features/promotion/applying_catalog_promotions/applying_catalog_promotion_with_proper_priorities.feature [hunk 1]
```diff
@@ -6,12 +6,18 @@ Feature: Applying catalog promotion with proper priorities
...
+        And there is a catalog promotion "Variant 10 sale" with priority 10 that reduces price by "30%" and applies on "PHP T-Shirt" variant
+        And there is a catalog promotion "Variant 100 sale" with priority 100 that reduces price by "50%" and applies on "PHP T-Shirt" variant
+        And there is a catalog promotion "Product 200 sale" with priority 200 that reduces price by "10%" and applies on "T-Shirt" product
+        And there is a catalog promotion "Product 20 sale" with priority 20 that reduces price by "10%" and applies on "T-Shirt" product
+        And there is a catalog promotion "Taxon 150 sale" with priority 150 that reduces price by "10%" and applies on "Clothes" taxon
+        And there is a catalog promotion "Taxon 15 sale" with priority 15 that reduces price by "10%" and applies on "Clothes" taxon
 
-    @api @ui
+    @ui
     Scenario: Applying catalog promotion with descending order by their priority
```
Comment (AdamKasp) [inline features/promotion/applying_catalog_promotions/applying_catalog_promotion_with_proper_priorities.feature:21]: of course I will cover it in API too

File: src/Sylius/Behat/Context/Ui/Shop/ProductContext.php [hunk 2]
```diff
@@ -510,9 +511,10 @@ public function iShouldSeeVariantIsDiscountedFromToWithPromotions(
 
         Assert::same($this->showPage->getPrice(), $price);
         Assert::same($this->showPage->getOriginalPrice(), $originalPrice);
-        foreach ($promotionsNames as $promotionName) {
-            Assert::true($this->showPage->hasCatalogPromotionApplied($promotionName));
-        }
+
+        $catalogPromotionNames = $this->showPage->getCatalogPromotionNames();
+        $catalogPromotionNames = array_map(fn (string $name) => explode(' - ', $name)[0], $catalogPromotionNames);
```
Comment (AdamKasp) [inline src/Sylius/Behat/Context/Ui/Shop/ProductContext.php:516]: TBH it is my first attempt to use the arrow function in PHP 👯

File: src/Sylius/Bundle/CoreBundle/Fixture/Factory/CatalogPromotionExampleFactory.php [hunk 3]
```diff
@@ -106,7 +106,8 @@ public function create(array $options = []): CatalogPromotionInterface
             }
         }
 
-        $this->catalogPromotionProcessor->process($catalogPromotion);
+        //it is probably redundant
+//        $this->catalogPromotionProcessor->process($catalogPromotion);
```
Comment (AdamKasp) [inline src/Sylius/Bundle/CoreBundle/Fixture/Factory/CatalogPromotionExampleFactory.php:110]: depending on the build color I will delete it or left.

**Comment by NoResponseMate** [comment]
Hi @AdamKasp 

I'm closing this since it has been done in 1.12
Cheers 🍻

**Pull request closed without merging**